airflow/providers/google/cloud/hooks/compute_ssh.py

```diff
@@ -0,0 +1,327 @@
+            self._sshclient = paramiko.SSHClient()
+            # Default is RejectPolicy
+            # No knownhost checking since we are not storing pkey
+            self._sshclient.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+            self._sshclient.connect(hostname=hostname,
+                                    username=username,
+                                    key_filename=private_key_file,
+                                    look_for_keys=False)
+        return self._sshclient
+
+    def get_service_account(self) -> str:
+        """
+        Get the service account id
+        """
+        account: str = self._get_credentials().service_account_email
+        os_login_client = self.get_oslogin_conn()
+        return os_login_client.user_path(user=account)
+
+    def get_username(self, oslogin=True) -> str:
+        """
+        Get the account username
+        """
+        account = self._get_credentials().service_account_email
+        if oslogin:
+            profile = self.get_oslogin_conn().get_login_profile(name=self.get_service_account())
+            account = profile.posix_accounts[0]
+            return account.username
+        return account.split("@")[0]
+
+    @GoogleBaseHook.fallback_to_default_project_id
+    def get_connection_type(self, project_id: Optional[str], instance: str, zone: str,
+                            hostname: Optional[str] = None,
+                            use_iap_tunnel: bool = False,
+                            use_internal_ip: bool = False
+                            ) -> str:
+        """
+        Selects how to connect to the target instance
+
+        :param project_id: The project ID of the instance
+        :type project_id: Optional[str]
+        :param instance: The name of the target instance to connect to
+        :type instance: str
+        :param zone: The zone of the target instance
+        :param zone: str
+        :param hostname: The hostname of the target instance. If provided, this would
+            be used, regardless of the setting for use_iap_tunnel and use_internal_ip
+        :type hostname: str
+        :param use_iap_tunnel: Whether to connect through IAP tunnel
+        :type use_iap_tunnel: bool
+        :param use_internal_ip: Whether to connect using internal IP
+        :type use_internal_ip: bool
+        """
+        if not hostname:
+            instance_info = self.get_conn() \
+                .instances().get(project=project_id,
+                                 zone=zone,
+                                 instance=instance).execute()
+            if use_iap_tunnel or use_internal_ip:
+                return instance_info["networkInterfaces"][0] \
+                    .get("networkIP")
+            else:
+                external_ip = instance_info \
+                    .get("networkInterfaces")[0] \
+                    .get("accessConfigs")
+                if external_ip:
+                    return external_ip[0].get("natIP")
```

Comment:
When natIP is empty, we should look at the access configuration.

Reply:
The natIP is inside the accessConfigs and when external IP is removed, the accessConfigs returns an empty list